apps/root/src/pages/home/components/portfolio/index.tsx

```diff
@@ -276,11 +296,59 @@ const Portfolio = ({ selectedWalletOption }: PortfolioProps) => {
       {}
     );
 
+    const allTokenKeys = Object.keys(tokenBalances);
+
+    // Merge multi-chain tokens
+    allTokenKeys.forEach((balanceItemKey) => {
+      const balanceItem = tokenBalances[balanceItemKey];
+      if (!balanceItem) {
+        return;
+      }
+
+      const { token: firstAddedToken } = balanceItem.tokens[0];
+
+      firstAddedToken.chainAddresses?.forEach(({ address, chainId }) => {
+        const equivalentItemKey = `${chainId}-${address}`;
+        const equivalentItem = tokenBalances[equivalentItemKey];
+        if (equivalentItem && equivalentItemKey !== balanceItemKey) {
+          balanceItem.isLoadingPrice = balanceItem.isLoadingPrice || equivalentItem.isLoadingPrice;
+          const equivalentToken = equivalentItem.tokens[0];
+          balanceItem.tokens.push(equivalentToken);
+
+          tokenBalances[balanceItemKey] = balanceItem;
+          delete tokenBalances[`${chainId}-${address}`];
+        }
+      });
+
+      if (balanceItem.tokens.length > 1) {
+        // All equivalent tokens should have same decimals, but we check just in case against the first token
+        const totalBalance = balanceItem.tokens.reduce(
+          (acc, { balance, token: equivalentToken }) =>
+            acc + parseUnits(balance.toString(), firstAddedToken.decimals - equivalentToken.decimals) || 0n,
```

Comment:
wait why are we doing this based on the difference of decimals? Shouldn't each balance be calculated by the tokens decimals of that balance?

Reply:
You're right, new approach in progress!